docs/elements.md

```diff
@@ -1652,19 +1651,47 @@ def generate(data):
     data["params"]["matrix"] = pl.to_json(mat)
 ```
 
+---
+
+**question.html**
+
+```html
+<pl-graph params-name-networkx="random-graph"></pl-graph>
+```
+
+**server.py**
+
+```python
+import prairielearn as pl
+import networkx as nx
+
+def generate(data):
+    random_graph = nx.gnm_random_graph(5, 6)
+
+    for in_node, out_node, edge_data in random_graph.edges(data=True):
+        edge_data["label"] = random.choice(string.ascii_lowercase)
+
+    data["params"]["random-graph"] = pl.to_json(random_graph)
+```
+
 #### Customizations
 
 | Attribute                   | Type    | Default            | Description                                                                                                                                                                                                                                                             |
 | --------------------------- | ------- | ------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | `engine`                    | string  | dot                | The rendering engine to use; supports `circo`, `dot`, `fdp`, `neato`, `osage`, and `twopi`.                                                                                                                                                                             |
-| `params-name-matrix`        | string  | `None`             | The the name of a parameter containing the adjacency matrix to use as input for the graph.                                                                                                                                                                              |
+| `params-name-matrix`        | string  | `None`             | The the name of a parameter containing the adjacency matrix to use as input if `params-type` is set to `adjacency-matrix`.                                                                                                                                              |
```

Comment:
@nwalters512 looking at this again, would you be open to consolidating the `params-name-matrix` and `params-name-graph` into a single attribute, and having legacy support for `params-name-matrix` (via a passthrough)? Just simplifies things and makes the parameter name less confusing.

Reply:
I think that'd be reasonable, yeah. Are you thinking that we'd just check the type of the (JSON-decoded) param and switch behavior based on that?

Reply:
Not even that complex, `params-type` is already explicitly given, which specifies the backend to use. Then `params-name` is just giving the entry in the data dictionary to pull from (to be interpreted by the backend however the backend wants to).